core/servers/chat/mrc_multiplexer.js

```diff
@@ -0,0 +1,325 @@
+const _                     = require('lodash');
+const os                    = require('os');
+
+
+// MRC
+const protocolVersion   = '1.2.9';
+const lineDelimiter     = new RegExp('\r\n|\r|\n');
+
+const ModuleInfo = exports.moduleInfo = {
+    name        : 'MRC',
+    desc        : 'An MRC Chat Multiplexer',
+    author      : 'RiPuk',
+    packageName : 'codes.l33t.enigma.mrc.server',
+    notes       : 'https://bbswiki.bottomlessabyss.net/index.php?title=MRC_Chat_platform',
+};
+
+const connectedSockets = new Set();
+
+exports.getModule = class MrcModule extends ServerModule {
+    constructor() {
+        super();
+
+        this.log            = Log.child( { server : 'MRC' } );
+        this.config         = Config();
+        this.mrcConnectOpts = {
+            host         : this.config.chatServers.mrc.serverHostname || 'mrc.bottomlessabyss.net',
+            port         : this.config.chatServers.mrc.serverPort || 5000,
+            retryDelay   : this.config.chatServers.mrc.retryDelay || 10000
+        };
+    }
+
+    _connectionHandler() {
+        const config = Config();
+        const boardName  = config.general.prettyBoardName || config.general.boardName;
+        const enigmaVersion = 'ENiGMA-BBS_' + require('../../../package.json').version;
+
+        const handshake = `${boardName}~${enigmaVersion}/${os.platform()}-${os.arch()}/${protocolVersion}`;
+        this.log.debug({ handshake : handshake }, 'Handshaking with MRC server');
+
+        this.mrcClient.write(handshake);
+        this.log.info(this.mrcConnectOpts, 'Connected to MRC server');
+    }
+
+    createServer(cb) {
+
+        if (!this.enabled) {
+            return cb(null);
+        }
+
+        this.connectToMrc();
+        this.createLocalListener();
+
+        return cb(null);
+    }
+
+    listen(cb) {
+        if (!this.enabled) {
+            return cb(null);
+        }
+
+        const config = Config();
+
+        const port = parseInt(config.chatServers.mrc.multiplexerPort);
+        if(isNaN(port)) {
+            this.log.warn( { port : config.chatServers.mrc.multiplexerPort, server : ModuleInfo.name }, 'Invalid port' );
+            return cb(Errors.Invalid(`Invalid port: ${config.chatServers.mrc.multiplexerPort}`));
+        }
+        Log.info( { server : ModuleInfo.name, port : config.chatServers.mrc.multiplexerPort }, 'MRC multiplexer local listener starting up');
+        return this.server.listen(port, cb);
+    }
+
+    /**
+     * Handles connecting to to the MRC server
+     */
+    connectToMrc() {
+        const self = this;
+
+        // create connection to MRC server
+        this.mrcClient = net.createConnection(this.mrcConnectOpts, self._connectionHandler.bind(self));
+
+        this.mrcClient.requestedDisconnect = false;
+
+        // do things when we get data from MRC central
+        var buffer = new Buffer.from('');
+
+        function handleData(chunk) {
+            if (typeof (chunk) === 'string') {
```

Comment:
ENiG requires & uses lodash all of the the place, so you can e.g. 
```
const _ = require('lodash');
// ...
if(_.isString(chunk)) { ... }
```

Reply:
oh, much nicer!